src/client/Client.cc

```diff
@@ -9018,8 +9018,7 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
     return r;
 
   /* clear the setuid/setgid bits, if any */
-  if (unlikely((in->mode & S_ISUID) ||
-	       (in->mode & (S_ISGID | S_IXGRP)) == (S_ISGID | S_IXGRP))) {
+  if (unlikely(in->mode & (S_ISUID|S_ISGID)) && size > 0) {
```

Comment:
According to [1] from the 1997 POSIX spec, this is the right behavior?

[1] http://pubs.opengroup.org/onlinepubs/7908799/xsh/write.html